README
NAME
    XML::Ant::BuildFile::Project - consume Ant build files

VERSION
    version 0.215

SYNOPSIS
        use XML::Ant::BuildFile::Project;

        my $project = XML::Ant::BuildFile::Project->new( file => 'build.xml' );
        print 'Project name: ', $project->name, "\n";
        print "File lists:\n";
        for my $list_ref (@{$project->file_lists}) {
            print 'id: ', $list_ref->id, "\n";
            print join "\n", @{$list_ref->files};
            print "\n\n";
        }

DESCRIPTION
    This class uses XML::Rabbit to consume Ant build files using a Moose
    object-oriented interface. It is a work in progress and in no way a
    complete implementation of all Ant syntax.

ATTRIBUTES
  file
    On top of XML::Rabbit's normal behavior, this class will also coerce
    Path::Class::File objects to the strings expected by
    XML::Rabbit::Role::Document.

  name
    Name of the Ant project.

  paths
    Hash of XML::Ant::BuildFile::Resource::Paths from the build file. The
    keys are the path "id"s.

  targets
    Hash of XML::Ant::BuildFile::Targets from the build file. The keys are
    the target names.

METHODS
  filelists
    Returns an array of all filelists in the project.

  filelist
    Given an index number returns that "filelist" from the project. You can
    also use negative numbers to count from the end. Returns "undef" if the
    specified "filelist" does not exist.

  map_filelists
    Given a code reference, transforms every "filelist" element into a new
    array.

  filter_filelists
    Given a code reference, returns an array with every "filelist" element
    for which that code returns "true".

  find_filelist
    Given a code reference, returns the first "filelist" for which the code
    returns "true".

  num_filelists
    Returns a count of all "filelist"s in the project.

  path
    Given a list of one or more "id" strings, returns a list of
    XML::Ant::BuildFile::Resource::Paths for "<classpath/>"s and "<path/>"s
    in the project.

  target
    Given a list of target names, return the corresponding
    XML::Ant::BuildFile::Target objects. In scalar context return only the
    last target specified.

  all_targets
    Returns a list of all targets as XML::Ant::BuildFile::Target objects.

  target_names
    Returns a list of the target names from the build file.

  has_target
    Given a target name, returns true or false if the target exists.

  num_targets
    Returns a count of the number of targets in the build file.

  BUILD
    After construction, the app-wide XML::Ant::Properties singleton stores
    any "<property/>" name/value pairs set by the build file, as well as any
    resource string expansions handled by XML::Ant::BuildFile::Resource
    plugins. It also contains the following predefined properties as per the
    Ant documentation:

    os.name
    basedir
    ant.file
    ant.project.name

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc XML::Ant::BuildFile::Project

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        The default CPAN search engine, useful to view POD in HTML format.

        <http://search.cpan.org/dist/XML-Ant-BuildFile>

    *   RT: CPAN's Bug Tracker

        The RT ( Request Tracker ) website is the default bug/issue tracking
        system for CPAN.

        <http://rt.cpan.org/NoAuth/Bugs.html?Dist=XML-Ant-BuildFile>

    *   AnnoCPAN

        The AnnoCPAN is a website that allows community annonations of Perl
        module documentation.

        <http://annocpan.org/dist/XML-Ant-BuildFile>

    *   CPAN Ratings

        The CPAN Ratings is a website that allows community ratings and
        reviews of Perl modules.

        <http://cpanratings.perl.org/d/XML-Ant-BuildFile>

    *   CPAN Forum

        The CPAN Forum is a web forum for discussing Perl modules.

        <http://cpanforum.com/dist/XML-Ant-BuildFile>

    *   CPANTS

        The CPANTS is a website that analyzes the Kwalitee ( code metrics )
        of a distribution.

        <http://cpants.perl.org/dist/overview/XML-Ant-BuildFile>

    *   CPAN Testers

        The CPAN Testers is a network of smokers who run automated tests on
        uploaded CPAN distributions.

        <http://www.cpantesters.org/distro/X/XML-Ant-BuildFile>

    *   CPAN Testers Matrix

        The CPAN Testers Matrix is a website that provides a visual way to
        determine what Perls/platforms PASSed for a distribution.

        <http://matrix.cpantesters.org/?dist=XML-Ant-BuildFile>

    *   CPAN Testers Dependencies

        The CPAN Testers Dependencies is a website that shows a chart of the
        test results of all dependencies for a distribution.

        <http://deps.cpantesters.org/?module=XML::Ant::BuildFile::Project>

  Bugs / Feature Requests
    Please report any bugs or feature requests through the web interface at
    <https://github.com/mjgardner/xml-ant-buildfile/issues>. You will be
    automatically notified of any progress on the request by the system.

  Source Code
    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    <https://github.com/mjgardner/xml-ant-buildfile>

      git clone git://github.com/mjgardner/xml-ant-buildfile.git

AUTHOR
    Mark Gardner <mjgardner@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by GSI Commerce.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
0.215     2011-05-23 14:53:01 America/New_York
    More complete file portability tests
0.214     2011-04-20 16:46:41 America/New_York
    MS Windows portability fixes in the tests
0.213     2011-04-20 15:32:55 America/New_York
    more support documentation
0.212     2011-04-20 13:27:45 America/New_York
    only require perl 5.10 instead of 5.12
0.211     2011-04-19 18:20:21 America/New_York
    added TaskContainer::find_task and ResourceContainer::map_resources
    delegated methods
    support <copy todir=""/> attribute
0.210     2011-04-19 15:02:08 America/New_York
    support <concat/> tasks and refid attributes in resources
    fix bug when default Ant properties are overridden with subsequent
    references to the main project
0.209     2011-04-19 11:33:22 America/New_York
    handle <arg pathref="foo"/> elements in <java/> tasks
0.208     2011-04-19 10:15:02 America/New_York
    require MooseX::Singleton 0.26 for compatibility with Moose >1.9900
0.207     2011-04-18 17:13:37 America/New_York
    fix fatal bug if a property value is undefined
0.206     2011-04-18 12:47:30 America/New_York
    better property and resource name substitution
    processing <java/> task arguments and <path/> structures
    new plugin system for XML::Ant::BuildFile::Resource and ::Task
0.205     2011-03-25 17:00:33 America/New_York
    new public apply_properties method on projects
    property substution is recursive
    rename Target and FileList classes up a level to XML::Ant::BuildFile::
    added XML::Ant::BuildFile::Task::Java and limited argument parsing
    lots of new helper methods for files, filelists, properties, targets, tasks
    targets, filelists, files automatically dereference
0.204     2011-03-24 16:16:11 America/New_York
    add some spiffy delegates and attributes for targets
0.203     2011-03-24 15:00:31 America/New_York
    turn targets into a hashref of objects so we can query their contents
0.202     2011-03-23 14:58:44 America/New_York
    require XML::Rabbit 0.0.4 explicitly so the FileList module will compile
0.201     2011-03-22 21:44:49 America/New_York
    remove dependency test on ant executable
0.200     2011-03-22 13:57:16 America/New_York
    extract from internal GSI project


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.010; }

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'consume Ant build files',
  'AUTHOR' => 'Mark Gardner <mjgardner@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Scalar::Util' => '0',
    'Test::More' => '0.94',
    'Test::Most' => '0',
    'perl' => '5.004'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.30'
  },
  'DISTNAME' => 'XML-Ant-BuildFile',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'XML::Ant::BuildFile',
  'PREREQ_PM' => {
    'Carp' => '0',
    'English' => '0',
    'Modern::Perl' => '0',
    'Module::Pluggable' => '0',
    'Moose' => '0',
    'Moose::Role' => '0',
    'Moose::Util::TypeConstraints' => '0',
    'MooseX::Has::Sugar' => '0',
    'MooseX::Has::Sugar::Minimal' => '0',
    'MooseX::Singleton' => '0.26',
    'MooseX::Types::Moose' => '0',
    'MooseX::Types::Path::Class' => '0',
    'Path::Class' => '0',
    'Readonly' => '0',
    'Regexp::DefaultFlags' => '0',
    'TryCatch' => '0',
    'XML::Rabbit' => '0.0.4',
    'XML::Rabbit::Node' => '0.0.4',
    'XML::Rabbit::RootNode' => '0',
    'namespace::autoclean' => '0',
    'utf8' => '0'
  },
  'VERSION' => '0.215',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



